src/Repository/EncryptionKeyGeneratingEventRepository.php

```diff
@@ -0,0 +1,42 @@
+<?php
+declare(strict_types=1);
+
+namespace MyOnlineStore\EventSourcing\Repository;
+
+use MyOnlineStore\EventSourcing\Aggregate\AggregateRootId;
+use MyOnlineStore\EventSourcing\Event\Stream;
+use MyOnlineStore\EventSourcing\Service\KeyGenerator;
+
+/**
+ * @final
+ */
+class EncryptionKeyGeneratingEventRepository extends EventRepositoryDecorator
```

Comment:
This is a clear case where `final` **should** be applied to make it solid. If you limit the responsibility of the event repository, this extended decorator pattern can dropped in favour of a cleaner solution